apis/v1beta1/gateway_types.go

```diff
@@ -69,49 +69,33 @@ type GatewaySpec struct {
 	// logical endpoints that are bound on this Gateway's addresses.
 	// At least one Listener MUST be specified.
 	//
-	// Each listener in a Gateway must have a unique combination of Hostname,
-	// Port, and Protocol.
+	// Each Listener in a set of Listeners (for example, in a single Gateway)
+	// MUST be _distinct_, in that a traffic flow MUST be able to be assigned to
+	// exactly one listener. (This section uses "set of Listeners" rather than
+	// "Listeners in a single Gateway" because implementations MAY merge configuration
+	// from multiple Gateways onto a single data plane, and these rules _also_
+	// apply in that case).
 	//
-	// Within the HTTP Conformance Profile, the below combinations of port and
-	// protocol are considered Core and MUST be supported:
+	// Practically, this means that each listener in a set MUST have a unique
+	// combination of Port, Protocol, and, if supported by the protocol, Hostname.
 	//
-	// 1. Port: 80, Protocol: HTTP
-	// 2. Port: 443, Protocol: HTTPS
+	// Some combinations of port, protocol, and TLS settings are are considered
+	// Core support and MUST be supported by implementations if they support the
+	// associated Route type:
```

Comment:
One thing I liked about the previous wording was that it was clearly tied to a conformance profile, so to be compliant with the HTTP conformance profile, you needed to support HTTP on 80 and HTTPS on 443. That left room for adding a new conformance profile that was for HTTPS-only implementations for example.

This updated wording makes it sound like it would be impossible to build a HTTPS-only implementation of Gateway API, and I don't think we really want to say that.

Reply:
I was trying to pivot this on supporting the associated Route type, but since we're still going by conformance profiles, I agree this should reference them.